tools/validate_reflection.py

```diff
@@ -0,0 +1,713 @@
+                    assert ignore_swap not in reflect_class.swapped, "Reflection for %s references field \"%s\" in %s  and %s " % (next_reflect_class, ignore_swap, ignore_str, swap_str)
+                    reflect_class.ignored.append(ignore_swap)
+                else:
+                    assert ignore_swap not in reflect_class.swapped, "Reflection for %s repeats %s \"%s\"" % (next_reflect_class, ignore_or_swap)
+                    assert ignore_swap not in reflect_class.ignored, "Reflection for %s references field \"%s\" in %s  and %s " % (next_reflect_class, ignore_swap, swap_str, ignore_str)
+                    reflect_class.swapped.append(ignore_swap)
+                debug("ignore or swap %s --> %s, ignored count=%s, swapped count=%s" % (next_reflect_class, ignore_swap, len(reflect_class.ignored), len(reflect_class.swapped)))
+                current = next_reflect_ignores_swaps.find(ignore_swap_match.group(0), current) + len(ignore_swap_match.group(0))
+            else:
+                break
+        
+
+    def add_field(self, reflect_class_name, field):
+        reflect_class = self.find_or_add(reflect_class_name)
+        assert field not in reflect_class.fields, "Reflection for %s repeats field \"%s\"" % (reflect_class_name, field)
+        reflect_class.fields.append(field)
+        debug("add_field %s --> %s" % (reflect_class_name, field))
+
+def replace_multi_line_comment(match):
+    all=match.group(1)
+    all=EmptyScope.strip_extra_pattern.sub("", all)
+    debug("multiline found=%s" % (all))
+    match=EmptyScope.ignore_swap_pattern.search(all)
+    if match:
+        ignore_or_swap = match.group(1) 
+        all = match.group(2)
+        debug("multiline %s now=%s" % (ignore_or_swap, all))
+        invalid_chars=EmptyScope.invalid_chars_pattern.search(all)
+        if invalid_chars:
+            for ic in invalid_chars.groups():
+                debug("invalid_char=%s" % (ic))
+            debug("WARNING: looks like \"%s\" is intending to %s, but there are invalid characters - \"%s\"" % (all, ignore_or_swap, ",".join(invalid_chars.groups())))
+            return ""
+        groups=re.findall(EmptyScope.multi_line_comment_ignore_swap_pattern, all)
+        if groups is None:
+            return ""
+        rtn_str="// %s " % (ignore_or_swap)
+        add=False
+        for group in groups:
+            debug("group=%s" % (group))
+            if group is None:
+                continue
+            if add:
+                rtn_str+=", "
+            rtn_str+="%s" % (group)
+            add=True
+        debug("multiline rtn_str=%s" % (rtn_str))
+        return rtn_str
+    
+    debug("multiline no match")
+    return ""
+
+def replace_line_comment(match):
+    all=match.group(0)
+    debug("singleline found=%s" % (all))
+    if EmptyScope.single_comment_ignore_swap_pattern.match(all):
+        return all
+    else:
+        return "\n"
+
+def validate_file(file):
+    f = open(file, "r")
+    contents = f.read()
```

Comment:
`contents = '\n' + f.read() # lazy fix for complex regex`

Strings being immutable, Python probably copies the entire file contents twice as it was.

Reply:
Ok